mesh/nurbs.cpp

```diff
@@ -45,6 +45,42 @@ KnotVector::KnotVector(int order, int NCP)
    knot = -1.;
 }
 
+KnotVector::KnotVector(int order, const Vector& intervals,
+                       const Array<int>& continuity )
+{
+   Order = order;
+   NumOfElements = intervals.Size();
+   const int num_knots = Order * continuity.Size() - continuity.Sum();
+   NumOfControlPoints = num_knots - Order - 1;
+   knot.SetSize(num_knots);
+   real_t accum = 0.0;
+   int iknot = 0;
+   for (int i = 0; i < intervals.Size(); ++i)
+   {
+      const int multiplicity = max(1, Order - continuity[i]);
+      for (int j = 0; j < multiplicity; ++j)
+      {
+         knot[iknot] = accum;
+         ++iknot;
+      }
+      accum += intervals[i];
+   }
+   if (intervals.Size() < continuity.Size())
+   {
+      const int multiplicity = max(1, Order - continuity[continuity.Size() - 1]);
+      for (int j = 0; j < multiplicity; ++j)
+      {
+         knot[iknot] = accum;
+         ++iknot;
+      }
+   }
+   else
+   {
+      MFEM_ABORT("The periodic case is not handled right now in KnotVector::KnotVector");
```

Comment:
Long line.

Reply:
I've broken this string literal into two lines in the latest version.